Sources/NIO/ChannelInvoker.swift

```diff
@@ -132,47 +84,82 @@ public protocol ChannelOutboundInvoker {
     var eventLoop: EventLoop { get }
 }
 
-/// Default implementations for `ChannelOutboundInvoker` methods. Each method that returns a `EventLoopFuture` will just do the following:
+/// Extra `ChannelOutboundInvoker` methods. Each method that returns a `EventLoopFuture` will just do the following:
 ///     - create a new `EventLoopPromise<Void>`
 ///     - call the corresponding method that takes a `EventLoopPromise<Void>`
 ///     - return `EventLoopPromise.futureResult`
 extension ChannelOutboundInvoker {
+
+    /// Register on an `EventLoop` and so have all its IO handled.
+    ///
+    /// - returns: the future which will be notified once the operation completes.
     public func register() -> EventLoopFuture<Void> {
```

Comment:
all of these should to take `file: StaticString = #file, line: UInt = #line` and pass them into the `newPromise` function so the 'leaked promise detection' will work with these guys